include/options.hxx

```diff
@@ -662,12 +666,12 @@ template<> void Options::assign<>(Tensor<BoutReal> val, const std::string source
 template <> inline bool Options::similar<BoutReal>(BoutReal a, BoutReal b) const { return fabs(a - b) < 1e-10; }
 
 /// Specialised as routines
-template <> std::string Options::as<std::string>(Mesh* mesh) const;
-template <> int Options::as<int>(Mesh* mesh) const;
-template <> BoutReal Options::as<BoutReal>(Mesh* mesh) const;
-template <> bool Options::as<bool>(Mesh* mesh) const;
-template <> Field2D Options::as<Field2D>(Mesh* mesh) const;
-template <> Field3D Options::as<Field3D>(Mesh* mesh) const;
+template <> std::string Options::as<std::string>(std::string&& similar_to) const;
+template <> int Options::as<int>(int&& similar_to) const;
+template <> BoutReal Options::as<BoutReal>(BoutReal&& similar_to) const;
+template <> bool Options::as<bool>(bool&& similar_to) const;
+template <> Field2D Options::as<Field2D>(Field2D&& similar_to) const;
+template <> Field3D Options::as<Field3D>(Field3D&& similar_to) const;
```

Comment:
I think I led you astray with the suggestion of `T&&`, as I was thinking you would need to modify `similar_to` inside `as`, for some reason. This should just be `const T&` (or by value for the simple types).